rs/messaging/src/routing/stream_handler.rs

```diff
@@ -792,13 +807,20 @@ impl StreamHandlerImpl {
         }
     }
 
+    /// Inducts a message into `state`. This may fail and log an error for cases where there is
+    /// nothing we can do about it; for other cases it may return `(StateError, RequestOrResponse)`
+    /// as it is returned by `ReplicatedState::push_input()` upon failure such that we may generate
+    /// a reject response or push a reject signal.
```

Comment:
I would try to be clearer about the 3 possible outcomes:
1. Message successfully inducted, returns `None`.
2. Message silently dropped (e.g. late best-effort response), returns `None`.
3. Failed to induct message (error or canister migrating), returns `(StateError, RequestOrResponse)`. Caller is expected to produce a reject response or a reject signal.